src/render/direct3d12/SDL_render_d3d12.c

```diff
@@ -69,7 +82,56 @@
     }
 #define D3D_CALL(THIS, FUNC, ...)     (THIS)->lpVtbl->FUNC((THIS), ##__VA_ARGS__)
 #define D3D_CALL_RET(THIS, FUNC, ...) (THIS)->lpVtbl->FUNC((THIS), ##__VA_ARGS__)
-#define D3D_GUID(X)                   &(X)
+#endif
+
+/*
+ * Older Windows SDK headers declare some d3d12 functions with the wrong function prototype.
+ * - ID3D12Heap::GetDesc
+ * - ID3D12Resource::GetDesc
+ * - ID3D12DescriptorHeap::GetDesc
+ * (and 9 more)
+ * */
+
+#if defined(__MINGW32__) || defined(SDL_PLATFORM_XBOXONE) || defined(SDL_PLATFORM_XBOXSERIES)
+/*
+ * MinGW has correct function prototypes in the vtables, but defines wrong functions
+ * Xbox just needs these macros defined as used below (because CINTERFACE doesn't exist)
+ */
+
+#undef ID3D12DescriptorHeap_GetCPUDescriptorHandleForHeapStart
+#define ID3D12DescriptorHeap_GetCPUDescriptorHandleForHeapStart(THIS, ...) \
+    D3D_CALL_RET(THIS, GetCPUDescriptorHandleForHeapStart, ##__VA_ARGS__);
+
+#undef ID3D12DescriptorHeap_GetGPUDescriptorHandleForHeapStart
+#define ID3D12DescriptorHeap_GetGPUDescriptorHandleForHeapStart(THIS, ...) \
+    D3D_CALL_RET(THIS, GetGPUDescriptorHandleForHeapStart, ##__VA_ARGS__);
+
+#undef ID3D12Resource_GetDesc
+#define ID3D12Resource_GetDesc(THIS, ...) \
+    D3D_CALL_RET(THIS, GetDesc, ##__VA_ARGS__);
+
+#elif !defined(D3D12_SDK_VERSION) || D3D12_SDK_VERSION < 4
```

Comment:
Yes, 3 is good. I think I got 4 from local (bad) mingw headers.

```suggestion
#elif !defined(D3D12_SDK_VERSION) || D3D12_SDK_VERSION < 3
```